rust/processor/src/processors/ans_processor.rs

```diff
@@ -296,95 +335,193 @@ fn parse_ans(
+            // 2. SetReverseLookupEvents: parse to get current_ans_primary_names
+            // 3. ANS V1 burn events: If we see a V1 burn event, we ignore the V1 wsc's in this transaction
+            for (event_index, event) in user_txn.events.iter().enumerate() {
+                if let Some(renew_name_event) =
+                    RenewNameEvent::from_event(event, &ans_v2_contract_address, txn_version)
+                        .unwrap()
+                {
+                    v2_renew_name_events.push(renew_name_event);
+                }
+                if let Some((current_ans_lookup, ans_lookup)) =
+                    CurrentAnsPrimaryName::parse_v2_primary_name_record_from_event(
+                        event,
+                        txn_version,
+                        event_index as i64,
+                        &ans_v2_contract_address,
+                        &all_current_ans_primary_names,
+                    )
+                    .unwrap()
+                {
+                    all_current_ans_primary_names
+                        .insert(current_ans_lookup.pk(), current_ans_lookup);
+                    all_ans_primary_names.push(ans_lookup);
+                }
+
+                if let Some(migration_burn_event) = MigrationBurnEvent::from_event(
+                    event,
+                    &ans_v1_creator_address,
+                    &ans_v2_migration_burn_address,
+                )
+                .unwrap()
+                {
+                    migration_burn_events.insert(migration_burn_event.burned_v1_token_name);
+                }
+            }
+
+            // Parse V1 ANS write set changes
+            for (wsc_index, wsc) in transaction_info.changes.iter().enumerate() {
+                match wsc.change.as_ref().unwrap() {
+                    // TODO: Don't index v1 changes if token burn detected
+                    WriteSetChange::WriteTableItem(table_item) => {
+                        if let Some((current_ans_lookup, ans_lookup)) =
+                            CurrentAnsLookup::parse_name_record_from_write_table_item_v1(
+                                table_item,
+                                &ans_v1_name_records_table_handle,
+                                txn_version,
+                                wsc_index as i64,
+                            )
+                            .unwrap_or_else(|e| {
+                                error!(
+                                    error = ?e,
+                                    "Error parsing ANS v1 name record from write table item"
+                                );
+                                panic!();
+                            })
+                        {
+                            // Don't index this v1 wsc if it's been migrated to v2
+                            if migration_burn_events.contains(&current_ans_lookup.token_name) {
+                                continue;
+                            }
```

Comment:
why not?

Reply:
When a v1 token migrates to v2, the data in the v1 table also changes (for example, the v1 token's target address gets unset [Example txn](https://explorer.aptoslabs.com/txn/663354522/changes?network=local)). So we don't want to index the v1 data if we detect it's been migrated, we only want to index the v2 data.